### ISSUE TITLE
`no-loss-of-precision` false positive

### ISSUE THREAD

**Issue by zloirock** [eslint/eslint#17492]
### Environment

Node version: 20.5.1
ESLint version: 8.47.0
Operating System: macOS 13.5.1

### What parser are you using?

Default (Espree)

### What did you do?

```js
const X = 255.10000610351562;
```

### What did you expect to happen?

No error.

### What actually happened?

> This number literal will lose precision at runtime

---

In all modern engines:
![image](https://github.com/eslint/eslint/assets/2213682/c9fa827b-673c-4900-b4f4-f19702eadf61)

**Comment by Rec0iL99** [comment]
Hi @zloirock, thanks for the issue. According to https://github.com/eslint/eslint/issues/15767#issuecomment-1092297546 and https://github.com/eslint/eslint/issues/15767#issuecomment-1092308580, the rule is correct in reporting the error.

```bash
> const X = 255.10000610351562;
undefined
> X.toPrecision(17);
'255.10000610351563'
> X === 255.10000610351563
true
```

pushed it forward for @mdjermanovic to verify.

**Comment by zloirock** [comment]
In my vision, that means that the logic of this method is incorrect - as the result of `Math.fround(255.1)` I have `255.10000610351562`, not `255.10000610351563` - how should I test this method with the usage of literal and without disabling this method?

**Comment by nzakas** [comment]
I think there's probably something not quite right in `no-loss-of-precision`, but having a hard time wrapping my head around it. @mdjermanovic can you comment?

**Comment by Rec0iL99** [comment]
@mdjermanovic friendly ping. Waiting for your feedback here. Thanks.

**Comment by fasttime** [comment]
@mdjermanovic I think your feedback is wanted here.

**Comment by zloirock** [comment]
Note that some future ES methods like https://github.com/tc39/proposal-number-fromstring imply consistency with `.toString` output and (most likely) will not accept strings like `255.10000610351563` as a valid number.

**Comment by Rec0iL99** [comment]
ping @mdjermanovic

**Comment by nzakas** [comment]
At least at this point there doesn't seem to be interest in addressing this, so closing. We can always revisit in the future as I think there is still something fishy about this rule.
